src/backends/perf_pt/perf_pt.c

```diff
@@ -137,10 +137,9 @@ stash_maps(pid_t pid, const char *map_filename)
     DEBUG("saving map to %s", map_filename);
     bool ret = true;
 
-    mode_t old_mode = umask(MAPS_MODE);
-
     char *cmd = NULL;
-    int res = asprintf(&cmd, "cp /proc/%d/maps %s", pid, map_filename);
+    int res = asprintf(&cmd, "cp /proc/%d/maps %s && chmod 600 %s",
```

Comment:
Ouch, getting the shell to deal with filenames is a security disaster waiting to happen (e.g. filenames with spaces). Unless we're 100% sure this code will be entirely killed in the next PR, I think we have to escape things properly.